src/main/java/manager/TaskManager.java

```diff
@@ -0,0 +1,177 @@
+                PrintManager.printBotExceptionMessage(de.getMessage());
+            }
+            fm.saveTasksToFile(taskList, taskCount);
+            break;
+        case EVENT_STRING:
+            try {
+                addTask(cmdArray[INTEGER_ONE], EVENT);
+            } catch (ArrayIndexOutOfBoundsException ae) {
+                PrintManager.printBotExceptionMessage("event task needs details");
+            } catch (DukeException de) {
+                PrintManager.printBotExceptionMessage(de.getMessage());
+            }
+            fm.saveTasksToFile(taskList, taskCount);
+            break;
+        case EMPTY_STRING:
+            pm.printBotStatusMessage("no command detected, please try again");
+            break;
+        default:
+            pm.printBotStatusMessage("invalid command, please try again");
+            break;
+        }
+    }
+
+    public Task getTaskType(String newTask, TaskType type) throws DukeException {
+        switch (type) {
+        case ADD:
+            return new Task(newTask);
+        case TODO:
+            return new Todo(newTask);
+        case DEADLINE:
+            String[] taskDetailsArray = getTaskNameAndTime(newTask, BY_STRING);
+            return new Deadline(taskDetailsArray[INTEGER_ZERO], taskDetailsArray[INTEGER_ONE]);
+        case EVENT:
+            String[] taskDetailsArray2 = getTaskNameAndTime(newTask, AT_STRING);
+            return new Event(taskDetailsArray2[INTEGER_ZERO], taskDetailsArray2[INTEGER_ONE]);
+        }
+        return null;
+    }
+
+    public String[] getTaskNameAndTime(String taskDetail, String delimiter)
+            throws DukeException, StringIndexOutOfBoundsException {
+        checkForDelimiter(taskDetail, delimiter);
+        int indexOfDelimiter = taskDetail.indexOf(delimiter);
+
+        checkEmptyDetails(indexOfDelimiter);
+        String detail = taskDetail.substring(INTEGER_ZERO, indexOfDelimiter - INTEGER_ONE).trim();
+        String datetime = taskDetail.substring(indexOfDelimiter + delimiter.length()).trim();
+
+        checkEmptyDateTime(datetime);
+        String[] taskDetailsArray = {detail, datetime};
+        return taskDetailsArray;
+    }
+
+    private void checkEmptyDetails(int indexOfDelimiter) throws DukeException {
+        if (indexOfDelimiter == INTEGER_ZERO) {
+            throw new DukeException("\nDuke: can't find details");
+        }
+    }
+
+    private void checkEmptyDateTime(String datetime) throws DukeException {
+        if (datetime.length() == INTEGER_ZERO) {
+            throw new DukeException("\nDuke: can't find datetime");
+        }
+    }
+
+    private void checkForDelimiter(String taskDetail, String delimiter) throws DukeException {
+        if (!taskDetail.contains(delimiter)) {
+            throw new DukeException(String.format("\nDuke: can't find %s", delimiter));
+        }
+    }
+
+    public void addTask(String newTask, TaskType type) throws DukeException {
+        Task typ = getTaskType(newTask, type);
```

Comment:
As with the previous comments, spelling out in full "type" is better than "typ".